app/code/Magento/Inventory/Model/Source/Validator/CodeValidator.php

```diff
@@ -41,6 +41,8 @@ public function validate(SourceInterface $source): ValidationResult
             $errors[] = __('"%field" can not be empty.', ['field' => SourceInterface::SOURCE_CODE]);
         } elseif (preg_match('/\s/', $value)) {
             $errors[] = __('"%field" can not contain whitespaces.', ['field' => SourceInterface::SOURCE_CODE]);
+        } elseif (preg_match('/\$[:]*{(.)*}/', $value)) {
+            $errors[] = __('"%field" can not contain invalid characters.', ['field' => SourceInterface::SOURCE_CODE]);
```

Comment:
Does a translation for this exist already?

Reply:
I don't think. Is there a more generic one that i should use instead? wasn't sure what the recommended way is.

Reply:
I can do that. Should i add it in the i18n/en_us.csv file?

Reply:
For a patch release we can't add new translations. If the next release isn't a patch then this is fine.

Reply:
Updated the message in the latest commit.